core/src/test/java/org/elasticsearch/search/child/ChildQuerySearchIT.java

```diff
@@ -1926,4 +1927,51 @@ public void testHasChildInnerQueryType() {
                 QueryBuilders.hasChildQuery("child-type", new IdsQueryBuilder().addIds("child-id"))).get();
         assertSearchHits(searchResponse, "parent-id");
     }
+
```

Comment:
agreed, that is better. We should just test the integration part here (whether non default similarity gets applied) and not whether right scores get computed.